### PR TITLE
Remove Validate implementation for elasticsearch Hosts config.

### LINKED ISSUE #3298: Fix flaky APIKeyTest test_invalidate_by_id
Looks like there is a new flaky test showing up:

**Build and Test / System and Environment Tests / test_apikey.APIKeyTest.test_invalidate_by_id** (from nosetests)

**Error Message**
```
{u'invalidated_api_keys': [], u'error_count': 0}
```

**Stacktrace**
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/go/src/github.com/elastic/apm-server/tests/system/test_apikey.py", line 90, in test_invalidate_by_id
    assert invalidated.get("invalidated_api_keys") == [apikey["id"]], invalidated
AssertionError: {u'invalidated_api_keys': [], u'error_count': 0}
```

### PR DESCRIPTION
**Pull request by jalvz**
Hopefully fixes #3298.

### CODE REVIEW

**Comment by jalvz** [comment]
This issue was most likely caused by https://github.com/elastic/apm-server/commit/55e407838017a12ad873603432ce24c2d65b466d#diff-51a8e951fca7351941d7f0436e972e0dR34

File: elasticsearch/config.go [hunk 1]
```diff
@@ -64,14 +63,6 @@ func DefaultConfig() *Config {
...
 type Hosts []string
 
-// Validate ensures Hosts instance has at least one entry
-func (h Hosts) Validate() error {
-	if len(h) == 0 {
-		return errInvalidHosts
-	}
-	return nil
-}
-
```
Comment (simitt) [inline elasticsearch/config.go]: Can you elaborate in which scenarios this `Validate` method is now called where `len(h)==0` would be ok?

Comment (jalvz) [reply elasticsearch/config.go]: I don't know (hence the "hopefully" above). This definitely fixes this error:
`./apm-server apikey create -E apm-server.api_key.elasticsearch.username=admin -E apm-server.api_key.elasticsearch.password=changeme`
```Error processing configuration: `Hosts` must at least contain one hostname accessing 'apm-server.api_key.elasticsearch.hosts' (source:'command line flag')
```

If you think it is not related, I can create a separate ticket, link it to it, and leave #3298 open.

**Comment by jalvz** [comment]
As per offline discussion, this is working as intended.

**Pull request closed without merging**